konfuzio_sdk/evaluate.py

```
@@ -437,3 +437,216 @@ def get_wrong_vertical_merge(self):
+                        fp += 1
+                    elif page_gt.is_first_page and not page_pr.is_first_page:
+                        fn += 1
+                    elif not page_gt.is_first_page and not page_pr.is_first_page:
+                        tn += 1
+            if tp + fp != 0:
+                precision = tp / (tp + fp)
+            else:
+                if self.allow_zero:
+                    precision = None
+                else:
+                    raise ZeroDivisionError(
+                        "TP and FP are zero, please specify allow_zero=True if you want precision to be None."
+                    )
+            if tp + fn != 0:
+                recall = tp / (tp + fn)
+            else:
+                if self.allow_zero:
+                    recall = None
+                else:
+                    raise ZeroDivisionError(
+                        "TP and FN are zero, please specify allow_zero=True if you want recall to be None."
+                    )
+            if precision + recall != 0:
+                f1 = 2 * precision * recall / (precision + recall)
+            else:
+                if self.allow_zero:
+                    f1 = None
+                else:
+                    raise ZeroDivisionError(
+                        "FP and FN are zero, please specify allow_zero=True if you want F1 to be None."
+                    )
+            self.evaluation_results_by_category['tp'][category.id_] = tp
+            self.evaluation_results_by_category['fp'][category.id_] = fp
+            self.evaluation_results_by_category['fn'][category.id_] = fn
+            self.evaluation_results_by_category['tn'][category.id_] = tn
+            self.evaluation_results_by_category['precision'][category.id_] = precision
+            self.evaluation_results_by_category['recall'][category.id_] = recall
+            self.evaluation_results_by_category['f1'][category.id_] = f1
+
+    def tp(self, search: Category = None) -> Union[int, dict]:
```

In all the methods that accept the search param, make it such that if an invalid Category is passed (which means a Category that is not in the Project), a descriptive exception is raised rather than the default KeyError from trying to find a dictionary key that does not exist